app/models/chat.py

```diff
@@ -0,0 +1,23 @@
+from typing import AsyncIterator
+from ._base import Computation
+from .matching import ScoredSentences
+from ..llm import LLMService
+
+class ChatModel(Computation):
+    def __init__(self, llm: LLMService):
+        self.llm = llm
+
+    async def compute(self, sentences: list[ScoredSentences], query: str) -> AsyncIterator[str]:
```

Comment:
scores are not needed here, rename `sentences` to `context`

Reply:
to do RAG, we should actually call the LLM with a prompt that contains the sentences we retrieved before as context

```
join_context = ".\n".join(sentences)
prompt = f"""
    You are giving the context and the question, try to generate the answer with the given context:
    context: {join_context}.
    question: {query}."""
```

Reply:
It's so cool, thanks for this comment 💯